File: frontend/src/admin/dto/PreRegisterUsersReportDto.ts
export interface PreRegisterUsersReportDto {
  totalUsers: number;
  updatedUsers: PreRegisterUserInternalDto[];
  invitedUsers: PreRegisterUserInternalDto[];
  failedUsers: Map<string, string>;
}

interface PreRegisterUserInternalDto {
  username: string;
}

File: backend/src/main/java/net/dnadas/training_portal/dto/user/PreRegisterUsersReportDto.java
package net.dnadas.training_portal.dto.user;

import java.util.List;
import java.util.Map;

public record PreRegisterUsersReportDto(
  Integer totalUsers, List<PreRegisterUserInternalDto> updatedUsers,
  List<PreRegisterUserInternalDto> invitedUsers,
  Map<String, String> failedUsers) {
}
